capsules/src/net/thread/tlv.rs

```diff
@@ -0,0 +1,1315 @@
+//! Implements Thread 1.1.1 Specification Type-Length-Value (TLV) encoding
+//! and decoding. Used during mesh link establishment (MLE), outlined in
+//! Chapter 4.
```

Comment:
Rather than just point to the specification (which isn't even available without signing their pseudo-NDA), summarize what TLVs are, what MLE is used for and high-level how you might use this module to build (one part) of a thread implementation.

Doesn't have to be 6 pages or anything, just an executive summary. Someone actually implementing thread will likely have to read the spec anyway, but think of the audience being someone who's _reading_ our thread implementation and browsing this module to understand the details. The code is there, so they can read it for themselves, but what background to they need to understand what the code is doing?